AnnoDesigner/UpdateHelper.cs

```diff
@@ -335,32 +335,71 @@ private AvailableRelease CheckForAvailableRelease(ReleaseType releaseType)
-            x.TagName.StartsWith(tagToCheck, StringComparison.OrdinalIgnoreCase));
-            //for testing - latest preset and icons release
-            //var latestPresetRelease = releases.FirstOrDefault(x => !x.Draft && !x.Prerelease && x.TagName.StartsWith("Presetsv3.0.0", StringComparison.OrdinalIgnoreCase));
-            if (foundGithubRelease == null)
+            Release foundPrerelease = null;
+            if (supportPrerelease)
             {
-                return result;
+                foundPrerelease = AllReleases.FirstOrDefault(x => !x.Draft && x.Prerelease && x.TagName.StartsWith(tagToCheck, StringComparison.OrdinalIgnoreCase));
+            }
+
+            Version versionPrerelease = default;
+            if (foundPrerelease != null)
+            {
+                versionPrerelease = ParseVersionFromTag(foundPrerelease, tagToCheck);
+                logger.Debug($"Found version (prerelease): {versionPrerelease}");
+            }
+
+            var foundRelease = AllReleases.FirstOrDefault(x => !x.Draft && !x.Prerelease && x.TagName.StartsWith(tagToCheck, StringComparison.OrdinalIgnoreCase));
+            Version versionRelease = default;
+            if (foundRelease != null)
+            {
+                versionRelease = ParseVersionFromTag(foundRelease, tagToCheck);
+                logger.Debug($"Found version: {versionRelease}");
             }
 
-            var versionString = foundGithubRelease.TagName.Replace(tagToCheck, string.Empty).Trim();
-            if (!Version.TryParse(versionString, out var foundVersion))
+            if (versionPrerelease == default && versionRelease == default)
             {
-                logger.Warn($"Could not get version of preset release. {nameof(foundGithubRelease.TagName)}: {foundGithubRelease.TagName}");
                 return result;
             }
 
-            result = new AvailableRelease
+            if (versionPrerelease != default && versionPrerelease > versionRelease)
+            {
+                result = new AvailableRelease
+                {
+                    Id = foundPrerelease.Id,
+                    Type = releaseType,
+                    Version = versionPrerelease
+                };
+            }
+            else
             {
-                Id = foundGithubRelease.Id,
-                Type = releaseType,
-                Version = foundVersion
-            };
+                result = new AvailableRelease
+                {
+                    Id = foundRelease.Id,
+                    Type = releaseType,
+                    Version = versionRelease
+                };
+            }
+
+            return result;
+        }
+
+        private Version ParseVersionFromTag(Release release, string tagToCheck)
+        {
+            var result = default(Version);
+
+            var versionString = release.TagName.Replace(tagToCheck, string.Empty).Trim();
+            if (!Version.TryParse(versionString, out var parsedVersion))
```

Comment:
Why don't we just use `result` here, rather than assigning a new variable? `default(Version)` will be `null`, which means we can just return `result` at the end of the method, and remove the `result = parsedVersion` line.

In fact - we could just get rid of `var result = default(Version)`  entirely, assuming `TryParse` does not set the out variable to an empty instance or something else other than `null`.

Reply:
As a matter of fact exactly this was my thought behind using a separate variable.
I don't know the inner of TryParse so I tried to make it explicit. Who cares about that extra bytes? It also is no critical path so there also is no GC pressure.